command-line-interface.md

```diff
@@ -0,0 +1,189 @@
+# Operations
+
+A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
```

Comment:
On Fri, Jul 29, 2016 at 06:00:08AM -0700, Doug Davis wrote:
> > @@ -0,0 +1,189 @@
> > +# Operations
> > +
> > +A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
>
> If we're in spec XXX it makes no sense to talk about people being
> compliant with spec YYY.

I think the current “protocols it implements” wording provides
sufficient granularity without having to define particular sub-specs
of runtime-spec [1].  *Every* MUST is effectively an independent
sub-spec.  So I don't see a problem with this file as it stands, or
with it after the changes I intend to make [1,2].

On the other hand, I haven't had much luck getting folks to agree with
my interpretation above.  The command line API is only weakly coupled
to the container configuration and container-process view of the
system.  So I'm also happy to have this command line API in a separate
repository [3].  Then the “if you want to be certified, here are the
APIs that the certification framework supports…” language [4] can just
live on the OCI-certification site.

So both of those approaches sound reasonable to me.  But I don't think
it makes sense to both land this API in runtime-spec *and* draw a
special box around it to mark it as “an independent spec” from “the
configuration spec”.  Either they're all part of runtime-spec, or we
should use separate repositories.  In either case, implementers can
pick and choose which protocols (i.e. portions of the spec) they want
to implement, and be compliant with those portions without having to
be compliant with the full spec.  The certification group will decide
which partial implementations are certifiable.

[1]: https://github.com/opencontainers/runtime-spec/pull/513/files#r72738319
[2]: https://github.com/opencontainers/runtime-spec/pull/513/files#r72713742
[3]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/BIxya5eSNLo
     Subject: Specifying the runtime's command-line interface
     Date: Tue, 18 Aug 2015 10:52:55 -0700
     Message-ID: <20150818175255.GV21585@odin.tremily.us>

     “If we do have a separate spec for the command-line API, I'd
     rather keep it in a separate repository…”
[4]: https://github.com/opencontainers/runtime-spec/pull/513/files#r72701237